dotCMS/src/main/java/com/dotcms/rest/api/v1/storage/FileStorageResource.java

```diff
@@ -0,0 +1,392 @@
+
+            contentlet = Try.of(()->this.contentletAPI.findContentletByIdentifier
+                    (identifier, mode.showLive, sessionLanguage, user, mode.respectAnonPerms)).getOrNull();
+        }
+
+        if (null == contentlet) {
+
+            final long defaultLanguage = APILocator.getLanguageAPI().getDefaultLanguage().getId();
+            if (defaultLanguage != sessionLanguage) {
+
+                contentlet =  Try.of(()->this.contentletAPI.findContentletByIdentifier
+                        (identifier, mode.showLive, defaultLanguage, user, mode.respectAnonPerms)).getOrNull();
+            }
+        }
+
+        return null == contentlet?
+                Optional.ofNullable(this.contentletAPI.findContentletByIdentifierAnyLanguage(identifier)):
+                Optional.ofNullable(contentlet);
+    }
+    /////////
+    @PUT
+    @Path("/metadata/get")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response getMetadata(@Context final HttpServletRequest request,
+                                      @Context final HttpServletResponse response,
+                                      final Map<String, Object> bodyMap)  {
+
+        if (UtilMethods.isSet(bodyMap)) {
+
+            final RequestMetaData.Builder builder = new RequestMetaData.Builder();
+            if (bodyMap.containsKey("cacheKey")) {
+
+                builder.cache(() -> (String) bodyMap.get("cacheKey"));
+            }
+
+            if (bodyMap.containsKey("file")) {
+
+                final String metadataBucketName = Config.getStringProperty("METADATA_BUCKET_NAME", "dotmetadata");
+                final Map fileMap = (Map) bodyMap.get("file");
+                final String inode       = (String)fileMap.get("id");
+                final String fileName    = (String)fileMap.get("name");
+                final String path        = (StringUtils.builder(File.separator,
+                        inode.charAt(0), File.separator, inode.charAt(1), File.separator, inode, File.separator,
+                        fileName).toString());
+                builder.storageKey(new StorageKey.Builder().bucket(metadataBucketName).path(path).build());
+            }
+
+            return Response.ok(new ResponseEntityView(this.fileStorageAPI.retrieveMetaData(builder.build()))).build();
+        }
+
+        return Response.status(Response.Status.BAD_REQUEST).build();
+    }
+
+    @PUT
+    @Path("/metadata/content/get")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response getContentMetadata(@Context final HttpServletRequest request,
+                                     @Context final HttpServletResponse response,
+                                     final MetadataForm metadataForm) throws DotSecurityException, DotDataException {
+
+        final InitDataObject initDataObject = new WebResource.InitBuilder(this.webResource)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)